### PR TITLE
update verifybamid output file name, nextflow verison use and workdir…

### PR DESCRIPTION
**Pull request by justinjj24**
- Update the verifybamid output file name to run multisample input

- Update nextflow version use compatible to this release `nextflowVersion = '!>=23.04.1'`

- Update work directory cleanup after successful run completion `cleanup = true`

### CODE REVIEW

**Comment by justinjj24** [comment]
Added `samtools stats` `--threads <>` argument to align with number of cpus in the `nextflow.config`

File: nextflow.config [hunk 1]
```diff
@@ -24,6 +25,9 @@ params {
     // Default to signed request when accessing s3 bucket
     // can be overwritten using params.aws_no_sign_request
     aws_no_sign_request = false
+
+    //To match samtools arg number of threads with no. of cpus
+    samtools_threads = '4'
```
Comment (mhebrard) [inline nextflow.config:30]: This should be a parameter extract from nextflow resource requirements, not hard coded

**Review by mhebrard — CHANGES_REQUESTED** [review]
* Add multi-thread option to samtools stats
* Fix verifybamid output filename
* Fix requirement for nextflow version
* Add nextflow cleanup uption (delete temporary files)
* Add NA12877 into test folder (to test multi-sample)

File: nextflow.config (hunk 1 above)
Comment (justinjj24) [reply nextflow.config:30]: Changed to argument..
```
        publishDir = [
            path: "${params.publishdir}/samtools",
```
```
nextflow run \
../../main.nf -config \
../../nextflow.config \
-params-file params.yml  \
-work-dir ./work  \
--publish_dir ./output \
--samtools_threads 2
```

File: nextflow.config [hunk 2]
```diff
@@ -33,7 +37,7 @@ process {
 
     withName: 'samtools_stats' {
 
-        cpus = 4
+        cpus = "${params.samtools_threads}"
```
Comment (nicolas-bertin) [inline nextflow.config:40]: keep a hard-coded value here. in this case `2`

File: nextflow.config [hunk 3]
```diff
@@ -24,6 +25,9 @@ params {
     // Default to signed request when accessing s3 bucket
     // can be overwritten using params.aws_no_sign_request
     aws_no_sign_request = false
+
+    //To match samtools arg number of threads with no. of cpus
+    samtools_threads = '2'
```
Comment (nicolas-bertin) [inline nextflow.config:30]: this should not be necessary

File: modules/samtools/main.nf [hunk 4]
```diff
@@ -11,6 +11,7 @@ process samtools_stats {
 
     script:
     def reference = ref_fasta ? /--reference "${ref_fasta}"/ : ''
+    def threads = "${params.samtools_threads}" ? /--threads "${params.samtools_threads}"/ : ''
```
Comment (nicolas-bertin) [inline modules/samtools/main.nf:14]: this should not be necessary

File: modules/samtools/main.nf [hunk 5]
```diff
@@ -19,6 +20,7 @@ process samtools_stats {
     samtools stats \\
         ${reference} \\
         "${bam}" \\
+        ${threads} \\
```
Comment (nicolas-bertin) [inline modules/samtools/main.nf:23]: should be `--threads ${task.cpus} `

File: nextflow.config (hunk 1 above)
Comment (justinjj24) [reply nextflow.config]: Added ${task.cpus}

**Review by mhebrard — APPROVED** [review]
* Add `tags` property into dockstore.yml
* Properly use `task.cpus` with samtools stats
* Update process resource requirements
* Enable wave

**Pull request merged**